### ISSUE TITLE
select-from-menu with a large input crashes StumpWM

### ISSUE THREAD

**Issue by nyx-land** [stumpwm/stumpwm#1028]
When trying to call `select-from-menu` with a large input list (hard for me to tell what the limit is but for me it's at around 2600), StumpWM will completely crash with a DRAWABLE-ERROR condition from CLX. I'm not sure how to properly debug this and provide more information since it's completely crashing and redirecting the debug output to a file doesn't seem to be catching anything relevant to this.

It can be reproduced by doing something like this:

```common-lisp
STUMPWM> (loop repeat 2600 for i = 0 then (incf i) collect (write-to-string i))
("0" "1" "2" "3" "4" "5" "6" "7" "8" "9" "10" "11" "12" "13" "14" "15" "16" ...)
STUMPWM> (select-from-menu (current-screen) *)
```

From what I can tell reading the stack trace after this crashes, xorg doesn't like that StumpWM attempts to write what is presumably too big a drawing for it to handle. I was trying to figure out how to make it so that when a menu is created, over a certain amount of entries it'll paginate instead of just writing everything to the screen, but I haven't figured out the StumpWM code well enough to know how to implement that.

Running this on FreeBSD 13.0 with the latest StumpWM release and SBCL 2.2.6

**Comment by sdilts** [comment]
Most of the code for menus can be found in [menu-declarations.lisp](https://github.com/stumpwm/stumpwm/blob/master/menu-declarations.lisp) and [menu-definitions.lisp](https://github.com/stumpwm/stumpwm/blob/master/menu-definitions.lisp). Menus should be at most `*maximum-menu-height*` entries tall, but by default that variable isn't set. What happens when you set `*menu-maximum-height*` and run the code you posted?

We might want to set the default value to something large but reasonable like 50 to avoid this problem from occurring.

**Comment by nyx-land** [comment]
I saw that but didn't realize it just wasn't set by default; that does indeed fix the problem! It's a trivial piece of code to change but if setting it to 50 by default seems reasonable I could make a PR for that change.

**Comment by sdilts** [comment]
That would be much appreciated!

**Comment by nyx-land** [comment]
Opened a PR for the fix: #1029
